neo4j/io/__init__.py

```diff
@@ -788,21 +788,21 @@ def _secure(s, host, ssl_context):
     local_port = s.getsockname()[1]
     # Secure the connection if an SSL context has been provided
     if ssl_context:
-        log.debug("[#%04X]  C: <SECURE> %s", local_port, host)
+        log.debug("[#%04X]  C: <ENCRYPTED> %s", local_port, host)
         try:
             sni_host = host if HAS_SNI and host else None
             s = ssl_context.wrap_socket(s, server_hostname=sni_host)
-        except SSLError as cause:
+        except (SSLError, OSError) as cause:
```

Comment:
What can trigger the OSError here?

Reply:
It is supposed to raise SSLError that is an subclass to OSError but somehow it does raise OSError.

> Raised to signal an error from the underlying SSL implementation (currently provided by the OpenSSL library). This signifies some problem in the higher-level encryption and authentication layer that’s superimposed on the underlying network connection. This error is a subtype of OSError. The error code and message of SSLError instances are provided by the OpenSSL library.
> 
> Changed in version 3.3: SSLError used to be a subtype of socket.error.

https://docs.python.org/3/library/ssl.html#ssl.SSLError

Reply:
So the OSError is triggered when connecting to a non secure connection.